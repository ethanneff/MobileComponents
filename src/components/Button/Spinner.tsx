import React from "react";
import { StyleSheet, View } from "react-native";
import { Circle } from "react-native-progress";
import Theme from "../../commons/Theme";

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
    alignItems: "center"
  }
});

interface Props {
  color?: string;
  hidden?: boolean;
}

class Spinner extends React.PureComponent<Props> {
  borderWidth = 2;
  size = Theme.padding.p5;
  render() {
    const { color = Theme.color.neutral900, hidden } = this.props;

    if (hidden) {
      return null;
    }
    return (
      <View style={styles.container}>
        <Circle
          borderWidth={this.borderWidth}
          color={color}
          size={this.size}
          indeterminate={true}
        />
      </View>
    );
  }
}

export default Spinner;
