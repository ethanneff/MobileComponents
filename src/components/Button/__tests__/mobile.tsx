import React from "react";
import renderer from "react-test-renderer";
import Component from "..";
import Theme from "../../../commons/Theme";
import Spinner from "../Spinner";

describe("Button Component", () => {
  it("renders correctly with title", () => {
    const dom = renderer
      .create(<Component title="press" onPress={() => undefined} />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with icon", () => {
    const dom = renderer
      .create(<Component icon="check" onPress={() => undefined} />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with icon title", () => {
    const dom = renderer
      .create(
        <Component icon="check" title="press" onPress={() => undefined} />
      )
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with loading", () => {
    const dom = renderer
      .create(<Component loading title="press" onPress={() => undefined} />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with secondary", () => {
    const dom = renderer
      .create(<Component secondary title="press" onPress={() => undefined} />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with primary", () => {
    const dom = renderer
      .create(<Component primary title="press" onPress={() => undefined} />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with background", () => {
    const dom = renderer
      .create(<Component background title="press" onPress={() => undefined} />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with half", () => {
    const dom = renderer
      .create(<Component half title="press" onPress={() => undefined} />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with disable", () => {
    const dom = renderer
      .create(<Component disable title="press" onPress={() => undefined} />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with nonFlex", () => {
    const dom = renderer
      .create(<Component nonFlex title="press" onPress={() => undefined} />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with clear", () => {
    const dom = renderer
      .create(<Component clear title="press" onPress={() => undefined} />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with circle", () => {
    const dom = renderer
      .create(<Component circle title="press" onPress={() => undefined} />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with fabCircle", () => {
    const dom = renderer
      .create(<Component fabCircle title="press" onPress={() => undefined} />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with bold", () => {
    const dom = renderer
      .create(<Component notBold title="press" onPress={() => undefined} />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with nonflex", () => {
    const dom = renderer
      .create(<Component nonFlex title="press" onPress={() => undefined} />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with alignLeft", () => {
    const dom = renderer
      .create(<Component title="press" onPress={() => undefined} alignLeft />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with alignRight", () => {
    const dom = renderer
      .create(<Component title="press" onPress={() => undefined} alignRight />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with invisible", () => {
    const dom = renderer
      .create(<Component title="press" onPress={() => undefined} invisible />)
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders correctly with styleSpinnerColor", () => {
    const dom = renderer
      .create(
        <Component
          styleSpinnerColor={Theme.color.brand}
          title="press"
          onPress={() => undefined}
        />
      )
      .toJSON();
    expect(dom).toMatchSnapshot();
  });
  it("renders hidden spinner", () => {
    const dom = renderer.create(<Spinner hidden />).toJSON();
    expect(dom).toMatchSnapshot();
  });
});
