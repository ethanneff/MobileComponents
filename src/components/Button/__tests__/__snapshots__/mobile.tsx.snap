// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly with alignLeft 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {
          "justifyContent": "flex-start",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with alignRight 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        Object {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with background 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(255, 100%, 100%)",
        "borderColor": "hsl(255, 100%, 100%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with bold 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            undefined,
            Object {
              "fontFamily": "BrandonText-Regular",
            },
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with circle 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      Object {
        "borderRadius": 80,
        "height": 56,
        "width": 56,
      },
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with clear 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            Object {
              "color": "hsl(255, 100%, 100%)",
            },
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with disable 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.5}
  disabled={true}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      undefined,
      Object {
        "opacity": 0.5,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with fabCircle 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      Object {
        "borderRadius": 48,
        "height": 64,
        "width": 64,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with half 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      Object {
        "alignSelf": "center",
        "width": "80%",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with icon 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "hsl(199, 18%, 33%)",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with icon title 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "hsl(199, 18%, 33%)",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with invisible 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      Object {
        "opacity": 0,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with loading 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      testID="mockProgressCircle"
    >
      <Text>
        Mock Progress Circle
      </Text>
      <Text
        testID="mockProgressCircleProps"
      >
        {"borderWidth":2,"color":"hsl(200, 19%, 18%)","size":20,"indeterminate":true}
      </Text>
    </View>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with nonFlex 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      Object {
        "alignSelf": "center",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with nonflex 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      Object {
        "alignSelf": "center",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with primary 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with secondary 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(204, 15%, 94%)",
        "borderColor": "hsl(204, 15%, 94%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with styleSpinnerColor 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders correctly with title 1`] = `
<TouchableOpacity
  accessibilityRole="button"
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 6,
        "justifyContent": "center",
        "padding": 12,
      },
      undefined,
      Object {
        "backgroundColor": "hsl(179, 58%, 67%)",
        "borderColor": "hsl(179, 58%, 67%)",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      style={
        Array [
          Array [
            Object {
              "color": "hsl(200, 19%, 18%)",
              "fontFamily": "BrandonText-Bold",
              "fontSize": 15,
              "letterSpacing": 0.5,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            undefined,
            Object {
              "fontFamily": "BrandonText-Bold",
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          undefined,
        ]
      }
      textBreakStrategy="simple"
    >
      PRESS
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Button Component renders hidden spinner 1`] = `null`;
